docker/dyno/app/api/docker.py

```diff
@@ -0,0 +1,228 @@
+    str : name
+        The name to normalize
+
+    Returns
+    -------
+    str
+        The normalized name
+    """
+    containers = container_list()
+    found = []
+    for container in containers['containers']:
+        if container.split('_').pop().strip() == name:
+            found.append(container)
+    if len(found) > 1:
+        raise Exception('Found more than one instance matching [{}]'.format(name))
+    if not found:
+        raise Exception('Could not normalize [{}] because it was not found in the container list')
+    return found[0]
+
+
+@bp.route('/list', methods=['GET'])
+def container_list():
+    """
+    Return list of containers
+
+    Note
+    ----
+    Exposed via HTTP at /api/docker/list
+
+    Note
+    ----
+    Paramaters are received query arguments in a Flask request object. They
+    may not be passed directly to this function.
+
+    Returns
+    -------
+    dict
+        A dict with a key of 'containers' and the value of which is a list of
+        the running containers
+
+    Examples
+    --------
+	❯ curl -s http://localhost:9000/api/docker/list|jq
+    {
+      "containers": [
+        "localtesting_8.0.0_opbeans-load-generator",
+        "localtesting_latest_opbeans-python",
+        "localtesting_latest_opbeans-python01",
+        "dyno",
+        "localtesting_8.0.0_apm-server",
+        "localtesting_8.0.0_kibana",
+        "localtesting_8.0.0_elasticsearch",
+        "localtesting_8.0.0_postgres",
+        "localtesting_8.0.0_redis",
+        "localtesting_8.0_toxi",
+        "localtesting_8.0_stats-d"
+      ]
+    }
+    """
+    ret = {'containers': []}
+    containers = client.containers.list()
+    for container in containers:
+        ret['containers'].append(container.name)
+    return ret
+
+
+@bp.route('/query', methods=['GET'])
+def query():
+    """
+    Inspect container and return information
+
+    Note
+    ----
+    Exposed via HTTP at /api/docker/query
+
+    Note
+    ----
+    Paramaters are received query arguments in a Flask request object. They
+    may not be passed directly to this function.
+
+    Parameters
+    ----------
+    c : str
+        Container to get
+
+    Returns
+    -------
+    dict
+        A dictionary showing the configuration for the running container
+
+    """
+    container = request.args.get('c')
+    config = low_client.inspect_container(_normalize_name(container))['HostConfig']
+    """
+    cpu_quota (int) - Limit CPU CFS (Completely Fair Scheduler) quota
+        -> HostConfig.CpuShares
+    blkio_weight (int) - Block IO (relative weight), between 10 and 1000
+        -> HostConfig.BlkioWeight
+    mem_limit (str) - Memory limit with units, such as 4m. The following
+    are supported: b, k, m, g
+        -> HostConfig.Memory
+    """
+    ret = {
+        'CPU': _denormalize_value('cpu', config['CpuQuota']),
+        'Mem': _denormalize_value('mem', config['Memory']),
+        # 'IO': _denormalize_value('io', config['BlkioWeight']),
```

Comment:
Is this somehow needed?

Reply:
This is just a reminder for possible future work. I couldn't get IO to work in my testing and I wanted to leave it in so I know where to put it in when I'm ready to come back to it. I will add a comment to make this more clear.